server/src/main/java/org/jboss/as/server/ElapsedTime.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.jboss.as.server;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+/**
+ * Tracks elapsed time since either JVM start or a moment of initialization. The start point
+ * for the calculation can be 'reset', allowing the calculation provided by an instance to
+ * account for a conceptual 'restart'.
```

Comment:
This needs updating as it describes an earlier form of behavior before I saw the need to switch to the 'checkpoint()' approach.